sapi/fpm/fpm/fpm_main.c

```diff
@@ -1996,7 +1996,8 @@ consult the installation file that came with this distribution, or visit \n\
 
 			requests++;
 			if (UNEXPECTED(max_requests && (requests == max_requests))) {
-				fcgi_finish_request(request, 1);
+				fcgi_request_set_keep(request, 0);
+				fcgi_finish_request(request, 0);
```

Comment:
Well, yeah, looks like. Although I don't use CGI SAPI and never looked into that code, so I can't really say.